hangman.py

```diff
@@ -0,0 +1,234 @@
+        print("Your total score for this game is:", final_points)
+    elif is_word_guessed(secret_word, letters_guessed) != 0 and guesses_point <= 0:
+        print("------")
+        print("Sorry, you ran out of guessed. The word was", secret_word)
+
+
+def hangman(secret_word, type_of_game):
+    '''
+    Game Hangman. "request" show hangman(classic) or hangman with hints.
+    Print information for user, start loop, check user choose hangman or hangman with hints,
+    Check input letter and call letter_in_word, check if user guessed word.
+    '''
+
+    guesses_point = 6
+    warnings_point = 3
+    letters_guessed = set()
+    print("I am thinking of a word that is", len(secret_word), "letters long")
+    print("You have", warnings_point, "warnings left")
+
+    while guesses_point > 0:
+        print("-------")
+        print("You have", guesses_point, "guesses left")
+        print("Available letters:", get_available_letters(letters_guessed))
+        letter = input("Please guess a letter:")
+        if not type_of_game and letter == GAME_WITH_HINTS:
+            print_hints_warning(secret_word, letters_guessed)
+        else:
+            warnings_point, guesses_point = check_letter(letter, warnings_point, guesses_point, secret_word, letters_guessed)
+        if is_word_guessed(secret_word, letters_guessed) == 0 or guesses_point <= 0:
+            break
+
+    #output result of game
+    result_of_game(secret_word, letters_guessed, guesses_point)
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise:
+    '''
+    if len(my_word) == len(other_word):
+        for i in range(len(my_word)):
+            if (my_word[i] != HIDDEN_LETTER and my_word[i] != other_word[i]) or (my_word[i] == HIDDEN_LETTER and other_word[i] in my_word):
+                return False
+        return True
+    else:
+        return False
+
+
+def show_possible_matches(my_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    returns: nothing, but should print out every word in wordlist that matches my_word
+             Keep in mind that in hangman when a letter is guessed, all the positions
+             at which that letter occurs in the secret word are revealed.
+             Therefore, the hidden letter(_ ) cannot be one of the letters in the word
+             that has already been revealed.
+    '''
+    list_with_words = []
+    my_word = my_word.replace(" ", "")
+    for word in list(wordlist):
+        if match_with_gaps(my_word, word) is True:
+            list_with_words.append(word)
+    return list_with_words
+
+
+def request_type():
```

Comment:
It's not obvious that `request_type` returns True or False. I'd expect it returns some type)
I'd rename it to ... `user_wants_hints`. Maybe it's a stupid example, but it still better than `request_type` I believe